File: 485-max-consecutive-ones/max-consecutive-ones.cs
public class Solution {
    public int FindMaxConsecutiveOnes(int[] nums) {
        int maxOnes = 0;
        int n = nums.Length;
        for(int i = 0; i < n; i++) {
            if(nums[i] != 1) continue;
            int cur = i, cnt = 0;
            while(cur < n && nums[i] == nums[cur]) {
                ++cur;
                ++cnt;
            }
            i = cur;
            maxOnes = Math.Max(maxOnes, cnt);
        }
        return maxOnes;
    }
}

File: 485-max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int n = nums.size();
        int prev = 0, maxOnes = 0, cnt = 1;
        for(int i = 0; i < n; i++) {
            if(nums[i] == 1) maxOnes = max(maxOnes, 1);
            if(prev == 1 and prev == nums[i]) {
                maxOnes = max(maxOnes, ++cnt);
            } else {
                cnt = 1;
            }
            prev = nums[i];
        }
        return maxOnes;
    }
};